src/track/trackref.h

```diff
@@ -0,0 +1,146 @@
+#ifndef TRACKREF_H
+#define TRACKREF_H
+
+
+#include <ostream>
+#include <utility>
+
+#include <QDateTime>
+#include <QFileInfo>
+#include <QString>
+#include <QtDebug>
+
+#include "track/trackid.h"
+#include "util/assert.h"
+
+
+// A track in the library is identified by a location and an id.
+// The location is mandatory to identify the file, whereas the id
+// only exists after the track has been inserted into the database.
+//
+// The logical reference to the track's file is referred to as its
+// "location". The location of each track must be unique inside the
+// library, i.e. each location identifies a single track. The location
+// contains an absolute path to the file.
+//
+// Different locations might reference the same file on disk. In order
+// to track which files are in use at runtime we also need to reference
+// tracks by their "canonical location". Each file has a unique canonical
+// location that is determined at runtime. The canonical location is
+// a transient property that may vary between different executions of the
+// application, but must not change during a single execution. It is
+// obtained by removing all redundancies from and resolving all symbolic
+// links in the location.
+//
+// NOTE(uklotzde): I've decided not to add any comparison operators for
+// this class, because technical and logical equality differs depending
+// on the use case!
+class TrackRef {
+public:
+    // Functions for extracting commonly used information from QFileInfo.
+    static QString toLocation(const QFileInfo& fileInfo) {
+        return fileInfo.absoluteFilePath();
+    }
+    static QString toCanonicalLocation(const QFileInfo& fileInfo) {
+        return fileInfo.canonicalFilePath();
+    }
+    static QString toDirectory(const QFileInfo& fileInfo) {
+        return fileInfo.absolutePath();
+    }
+    static QString toFileName(const QFileInfo& fileInfo) {
+        return fileInfo.fileName();
+    }
+    static quint64 toFileSize(const QFileInfo& fileInfo) {
+        return fileInfo.size();
+    }
+    static QDateTime toFileCreatedAt(const QFileInfo& fileInfo) {
+        return fileInfo.created();
+    }
+    static QDateTime toFileLastModifiedAt(const QFileInfo& fileInfo) {
+        return fileInfo.lastModified();
+    }
```

Comment:
Do you have a special plan for the wrapper functions above? 
IMHO the only hide implementation details from the calling code. 
I would perefer to call: 
fileInfo.fileName(); 
instead of 
TrackRef::fileName(fileInfo);